zarr/tests/test_storage.py

```diff
@@ -1337,6 +1337,24 @@ def test_s3_complex(self):
         )
         assert (a[:] == -np.ones((8, 8, 8))).all()
 
+    def test_exceptions(self):
+        import fsspec
+
+        m = fsspec.filesystem("memory")
+        g = zarr.open_group("memory://test/out.zarr", mode="w")
+        arr = g.create_dataset("data", data=[1, 2, 3, 4], dtype="i4", compression=None, chunks=[2])
+        m.store["/test/out.zarr/data/0"] = None
+        del m.store["/test/out.zarr/data/1"]
+        assert g.store.getitems(["data/1"], contexts={}) == {}  # not found
+        with pytest.raises(Exception):
+            # None is bad data, as opposed to missing
+            g.store.getitems(["data/0", "data/1"], contexts={})
+        with pytest.raises(Exception):
+            # None is bad data, as opposed to missing
+            arr[:]
+        # clear the global memory filesystem's store for use in other tests
+        m.store.clear()
```

Comment:
This seems somewhat problematic. Is there a better way to isolate this filesystem, perhaps by moving the filesystem (`m`) to a fixture? The concern here is that if this test fails before this step, the clean up will not happen.

Reply:
Resolved with fixture, thanks for the suggestion.